thrust/iterator/detail/tagged_iterator.h

```diff
@@ -58,6 +58,21 @@ template<typename Iterator, typename Tag>
       : super_t(x) {}
 }; // end tagged_iterator
 
+/*! \p make_tagged_iterator creates a \p tagged_iterator
+ *  from a \c Iterator with system tag \c Tag.
+ *
+ *  \tparam Tag Any system tag.
+ *  \tparam Iterator Any iterator type.
+ *  \param iter The iterator of interest.
+ *  \return An iterator whose system tag is \p Tag and whose behavior is otherwise
+ *          equivalent to \p iter.
+ */
+template <typename Tag, typename Iterator>
+inline auto make_tagged_iterator(Iterator iter)
```

Comment:
Shouldn't these be exposed outside of `detail::`?

Reply:
This can be done as a follow if needed.

Reply:
This will need to be updated with a trailing return type to support C++11:

```
/dvs/p4/build/sw/gpgpu/thrust/thrust/iterator/detail/tagged_iterator.h(71): error #2730-D: deduced return types are a C++14 feature
```

Reply:
```suggestion
inline auto make_tagged_iterator(Iterator iter) -> tagged_iterator<Iterator, Tag>
```